src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java

```diff
@@ -116,16 +120,40 @@ private List<UnknownWord> selectWords(UUID userId, int selectSize) throws Except
             selectSize = initialSelectSize - selectedWords.size();
             confidences.clear();
             confidences.addAll(Arrays.asList(ConfidenceEnum.HIGH, ConfidenceEnum.HIGHEST));
-            
-            selectedWords.addAll(unknownWordRepository.findRandomByOwnerListUserIdAndOwnerListIsActiveAndConfidence(
-                userId, true, confidences, selectSize));
+
+
+            if (selectedWords.size() == initialSelectSize) {
+                selectedWords.addAll(unknownWordRepository.findRandomByOwnerListUserIdAndOwnerListIsActiveAndConfidence(
+                        userId, true, confidences, selectSize));
+                return selectedWords;
+            }
+
+            log.error("Defaulting to the predefined word list, as selected words size is not equal to initial select size, " +
+                    "selectedWords size: {}, initialSelectSize: {}", selectedWords.size(), initialSelectSize);
+
+            QPredefinedWordList predefinedWordList = readPredefinedWordListFromYamlFile(DEFAULT_WORD_LIST_FILE);
```

Comment:
I'm just curious, is reading from the .yaml file each time more performant, or would it be better to store these predefined words in a new DB table and fetch from there?